### DISCUSSION TITLE
flowbite with react interactive elements don't work

### DISCUSSION THREAD

**Discussion by dawit-michael** [themesberg/flowbite#49] [Q&A] [answered]
I've tried to use flowbite via NPM. followed the quick start guide . I've installed the package + added to tailwind.config.js also added in the index.html  + imported flowbite in app.tsx. but the interactive elements still don't work :(

**Comment by zoltanszogyenyi** [comment, marked as answer]
Hey @dawit-michael,

We are aware of the issue of the event listeners not registering because of the virtual DOM from React. We're currently working on a compatibility solution for v1.2.1 which we could launch in a few days/early January based on progress.

Either way, we will launch a separate React + Flowbite library in February and that will make things a lot smoother.

Contributions are welcome and if you find a solution, let us know!

**Reply by dawit-michael** [reply]
I thought it was an issue of mine. Thank you for letting me know. Will keep you posted if I find anything. Thanks

**Reply by fwahlqvist** [reply]
Having the same issue with Gatsby, if anyone have got it to work or any update on the react lib would be greatly appriciated

**Reply by gcolliso** [reply]
Any update on getting the Flowbite interactive items to work with React/Next.js?

**Reply by bahaeben** [reply]
is there any news, i couldn't get interactive elements to work

**Reply by Aqibrafiq001** [reply]
is there any update ? im still facing same issue in v1.4.5

**Reply by zoltanszogyenyi** [reply]
Hey, everyone!

We have a standalone React library almost ready for production: https://github.com/themesberg/flowbite-react

Please check this out if you want to use it for React, Next JS or React Native.

Thanks!

**Reply by Aqibrafiq001** [reply]
flowbite is now running fine after adding cdn in index.html in reactjs.
please mention clearly in react documentation.

for those who are getting error you can do these changes
* add this line in head    <link rel="stylesheet" href="https://unpkg.com/flowbite@1.4.5/dist/flowbite.min.css" />
*  add this line in body     <script src="https://unpkg.com/flowbite@1.4.5/dist/flowbite.js"></script>

**Reply by Aqibrafiq001** [reply]
@zoltanszogyenyi please mention how to import component im getting error in importing in react i follow above link

**Reply by nathaniel-ross** [reply]
@Aqibrafiq001 what line do you need to add in head?

**Reply by skandarajeev** [reply]
add what in head, you left that blank

**Comment by adriDiazz** [comment]
> I've tried to use flowbite via NPM. followed the quick start guide . I've installed the package + added to tailwind.config.js also added in the index.html + imported flowbite in app.tsx. but the interactive elements still don't work :(

Hi have you solve the problem, im getting the same error

**Reply by stefanovolpatti** [reply]
Hi @adriDiazz
You can find the repo for React [here](https://github.com/themesberg/flowbite-react)

**Reply by Aqibrafiq001** [reply]
@volp99  how to import a component please let me know

**Reply by Aqibrafiq001** [reply]
@adriDiazz  still getting import issue and i didnt get any documentation on there flowbite-react

**Reply by DanielHemmati** [reply]
same problem

**Reply by dawit-michael** [reply]
@DanielCodex  have you followed through the [GETTING STARTED](https://flowbite.com/docs/getting-started/react/) for react. and tried using the [components](https://flowbite-react.com/). If possible can you provided further elaboration on the errors you're facing.. possibly screenshots?

**Reply by DanielHemmati** [reply]
here you go https://imgur.com/a/gYd2e1O
i did install it correctly, as you can see in the image for something like Alert component works but Navbar components doesn't work. i don't know how can i import something like `<Navbar.Link>`

**Reply by dawit-michael** [reply]
> here you go https://imgur.com/a/gYd2e1O i did install it correctly, as you can see in the image for something like Alert component works but Navbar components doesn't work. i don't know how can i import something like `<Navbar.Link>`


![image](https://user-images.githubusercontent.com/32758991/200019422-938b7b1b-806b-47e7-87e6-44ff54e53376.png)

**Error:  DEFINED MULTIPLE TIMES** _translation : you have two components using the same name_  If you see the marked images the problem is you have two components using the same name.....You should name your component to something other than Navbar because the compiler is going to have issue when referencing two components of same name.

**Reply by DanielHemmati** [reply]
why i didn't saw that 😅 
thanks for the help

**Reply by Namnika** [reply]
Hey, @DanielCodex, @dawit-michael  I was also getting that error earlier and I saw that multiple times  `<Navbar>` I defined, after getting solved, now, I'm getting this new error, Here you can see the screenshot: 

![Screenshot (218)](https://user-images.githubusercontent.com/92422534/222721220-364818a5-e0ab-40af-a171-ffafb159df7a.png)

I saw **getting-started** docs of **flowbite-react** but to add the navbar react component, still, I'm getting **_Module not found Error_**

 Please Help me with this error :(

**Reply by Namnika** [reply]
Hi @Aqibrafiq001, You're almost correct, they should mention in the docs how to import flowbite-react components after installing it :(

**Reply by dawit-michael** [reply]
@Namnika can you please share your Header.js code. and please try restarting.

**Reply by Namnika** [reply]
Hi @dawit-michael, Thank you so much for your quick response, 
I shared all relevant codes to try to find out errors easily as for your reference, 

This is my Header.js file code: 
```

`import React, { useState } from "react";
import { Navbar } from "flowbite-react";
import "./navbar.css";
import Switcher from "../Switcher";

export default function Header() {
  const [isNavExpanded, setIsNavExpanded] = useState(false);

  return (
    <>
      {/* I just only want img with brand name part in my navbar */}
      <div className="w-4/5">
        <Navbar fluid={true} rounded={true}>
          <React.Fragment key=".0">
            <Navbar.Brand href="https://flowbite.com/">
              <img
                src="https://flowbite.com/docs/images/logo.svg"
                className="mr-3 h-6 sm:h-9"
                alt="Flowbite Logo"
              />
              <span className="self-center whitespace-nowrap text-xl font-semibold dark:text-white">
                Flowbite
              </span>
            </Navbar.Brand>
          </React.Fragment>
        </Navbar>
      </div>
    </>
  );
}
`
```

this is package.json:

`"dependencies": {
    "@testing-library/jest-dom": "^5.16.5",
    "@testing-library/react": "^13.4.0",
    "@testing-library/user-event": "^13.5.0",
    "axios": "^1.3.3",
    "flowbite": "^1.6.3",
    "flowbite-react": "^0.4.1",
    "react": "^18.2.0",
    "react-dom": "^18.2.0",
    "react-scripts": "5.0.1",
    "react-toggle-dark-mode": "^1.1.1",
    "web-vitals": "^2.1.4"
  },`

this is tailwind.config.css:

`/** @type {import('tailwindcss').Config} */
module.exports = {
  darkMode: "class",
  content: ["./src/**/*.{js,jsx,ts,tsx}" ,"./node_modules/flowbite/**/*.js"],
  theme: {},
  plugins: [require("flowbite-typography"), require("flowbite/plugin")],
};
`
this is index.html :

```
<link href="https://cdnjs.cloudflare.com/ajax/libs/flowbite/1.6.3/flowbite.min.css" rel="stylesheet" />
    <title>React App</title>
  </head>
  <body>
    <div id="root"></div>
    <script src="https://cdnjs.cloudflare.com/ajax/libs/flowbite/1.6.3/flowbite.min.js"></script>
  </body>
```

I hope you will get this.. actually, I restarted it but still getting that error, Even then I tried to remove the navbar component and try to use the `<Footer>`  component but still unsuccessful :(

**Reply by dawit-michael** [reply]
In your tailwind.config.js you should add the following according to [this link](https://flowbite.com/docs/getting-started/react/)
` content: [
        // ...
        'node_modules/flowbite-react/**/*.{js,jsx,ts,tsx}'
    ]`
Also try to install the package again following the link [here](https://flowbite.com/docs/getting-started/react/)

**Reply by Namnika** [reply]
Thank you so muchh for helping me a lottt!!! 😭😭
When you said for restarting I restarted then it works better as expected!!
Actually, I was installing packages while running the localhost 😕

Here you can see it's working : 

![Screenshot (219)](https://user-images.githubusercontent.com/92422534/222758074-24d0387f-c0cd-49cb-95a1-9edfe79c8f4a.png)

Thanks a lottt!! :(

**Comment by gedeonagmas** [comment]
and for tooltip, add the following lines in your index.html file
`<body>
    <div id="root"></div>
    <script type="module" src="/src/main.jsx"></script>
    <script src="https://cdnjs.cloudflare.com/ajax/libs/flowbite/2.3.0/flowbite.min.js"></script>
    <script src="../path/to/flowbite/dist/flowbite.min.js"></script>
  </body>`
